src/react/hooks/paragon/utils.js

```diff
@@ -0,0 +1,101 @@
+  if (!themeVariants) {
+    return undefined;
+  }
+
+  const themeVariantKeys = Object.keys(themeVariants);
+
+  // Return early if there are no theme variants configured.
+  if (themeVariantKeys.length === 0) {
+    return undefined;
+  }
+  // If there is only one theme variant, return it since it's the only one that may be used.
+  if (themeVariantKeys.length === 1) {
+    const themeVariantKey = themeVariantKeys[0];
+    return {
+      name: themeVariantKey,
+      metadata: themeVariants[themeVariantKey],
+    };
+  }
+  // There's more than one theme variant configured; figured out which one to display based on
+  // the following preference rules:
+  //   1. Get theme preference from localStorage.
+  //   2. Detect user system settings.
+  //   3. Use the default theme variant as configured.
+
+  // Prioritize persisted localStorage theme variant preference.
+  const persistedSelectedParagonThemeVariant = localStorage.getItem(SELECTED_THEME_VARIANT_KEY);
+  if (persistedSelectedParagonThemeVariant && themeVariants[persistedSelectedParagonThemeVariant]) {
+    return {
+      name: persistedSelectedParagonThemeVariant,
+      metadata: themeVariants[persistedSelectedParagonThemeVariant],
+    };
+  }
+
+  // Then, detect system preference via `prefers-color-scheme` media query and use
+  // the default dark theme variant, if one exists.
+  const hasDarkSystemPreference = !!window.matchMedia?.('(prefers-color-scheme: dark)')?.matches;
+  const defaultDarkThemeVariant = themeVariantDefaults.dark;
+  const darkThemeVariantMetadata = themeVariants[defaultDarkThemeVariant];
+
+  if (hasDarkSystemPreference && defaultDarkThemeVariant && darkThemeVariantMetadata) {
+    return {
+      name: defaultDarkThemeVariant,
+      metadata: darkThemeVariantMetadata,
+    };
+  }
+
+  const defaultLightThemeVariant = themeVariantDefaults.light;
+  const lightThemeVariantMetadata = themeVariants[defaultLightThemeVariant];
+
+  // Handle edge case where the default light theme variant is not configured or provided.
+  if (!defaultLightThemeVariant || !lightThemeVariantMetadata) {
+    return undefined;
+  }
+
+  // Otherwise, fallback to using the default light theme variant as configured.
+  return {
+    name: defaultLightThemeVariant,
+    metadata: lightThemeVariantMetadata,
+  };
+};
+
+/**
+* Creates the fallback URL for the given theme file.
+* @param {string} url The theme file path.
+* @returns {string} The default theme url.
+*/
+export const fallbackThemeUrl = (url) => {
+  const baseUrl = getConfig().BASE_URL || window.location?.origin;
+  return `${baseUrl}${basename}${url}`;
+};
+
+export const handleVersionSubstitution = ({ url, wildcardKeyword, localVersion }) => {
+  if (!url || !url.includes(wildcardKeyword) || !localVersion) {
+    return url;
+  }
+  return url.replaceAll(wildcardKeyword, localVersion);
+};
+
+export const isEmptyObject = (obj) => !obj || Object.keys(obj).length === 0;
```

Comment:
It looks like someone did some investigation into performance of different ways to test for empty objects in js (https://stackoverflow.com/a/59787784) and found that using for-in is the fastest

 ```suggestion
export const isEmptyObject = (obj) => {
  for (var i in obj) {
    return false;
  }
  return true;
}
```

I don't think this is a required change at all, more just something fun I found when looking into it!